bids-validator/src/schema/applyRules.ts

```diff
@@ -168,61 +168,123 @@ function schemaObjectTypeCheck(
   schema: GenericSchema,
 ): boolean {
   // always allow n/a?
-  if (value === 'n/a') {
-    return true
+  if (value === "n/a") {
+    return true;
   }
-  if ('anyOf' in schemaObject) {
+  if ("anyOf" in schemaObject) {
     return schemaObject.anyOf.some((x) =>
-      schemaObjectTypeCheck(x, value, schema),
-    )
+      schemaObjectTypeCheck(x, value, schema)
+    );
   }
-  if ('enum' in schemaObject && schemaObject.enum) {
-    return schemaObject.enum.some((x) => x === value)
+  if ("enum" in schemaObject && schemaObject.enum) {
+    return schemaObject.enum.some((x) => x === value);
   }
   // @ts-expect-error
-  const format = schema.objects.formats[schemaObject.type]
-  const re = new RegExp(`^${format.pattern}$`)
-  return re.test(value)
+  const format = schema.objects.formats[schemaObject.type];
+  const re = new RegExp(`^${format.pattern}$`);
+  return re.test(value);
+}
+
+/**
+ * Checks user supplied type information from a sidecar against tsv column value.
+ */
+function sidecarDefinedTypeCheck(
+  rule: object,
+  value: string,
+  schema: GenericSchema,
+): boolean {
+  if (
+    "Levels" in rule && rule["Levels"] && typeof (rule["Levels"]) == "object"
+  ) {
+    return value in rule["Levels"];
+  }
+  if ("Units" in rule) {
+    schemaObjectTypeCheck({ "type": "number" }, value, schema);
+  }
+  return true;
```

Comment:
I think this is missing an `n/a` exception for `Levels`, and you aren't returning the value of the schemaObjectTypeCheck. I think if we use an `if/else if/else` block, typescript would be able to detect bad fallthrough behavior.

```suggestion
  if (
    "Levels" in rule && rule["Levels"] && typeof (rule["Levels"]) == "object"
  ) {
    return value == 'n/a' || value in rule["Levels"];
  } else if ("Units" in rule) {
    return schemaObjectTypeCheck({ "type": "number" }, value, schema);
  } else {
    return true;
  }
```